go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -265,10 +255,16 @@ func (oc *Controller) deleteEgressFirewall(egressFirewallObj *egressfirewallapi.
 		}
 	}
 	if deleteDNS {
-		oc.egressFirewallDNS.Delete(egressFirewallObj.Namespace)
+		if _, err := oc.egressFirewallDNS.Delete(egressFirewallObj.Namespace); err != nil {
+			return err
+		}
 	}
 
-	return oc.deleteEgressFirewallRules(egressFirewallObj.Namespace)
+	if err := oc.deleteEgressFirewallRules(egressFirewallObj.Namespace); err != nil {
+		return err
+	}
+	oc.egressFirewalls.Delete(egressFirewallObj.Namespace)
+	return nil
 }
 
 func (oc *Controller) updateEgressFirewallWithRetry(egressfirewall *egressfirewallapi.EgressFirewall) error {
```

Comment:
Can we call this `updateEgressFirewallStatusWithRetry` to make it easy to understand it on the fly without having to inspect it each time?